src/test/objectstore/test_deferred.cc

```diff
@@ -17,8 +22,38 @@
 #include "common/errno.h"
 #include "common/options.h" // for the size literals
 #include <semaphore.h>
+#include "os/bluestore/Allocator.h"
+#include "os/bluestore/AvlAllocator.h"
+
+class TweakedAllocator : public AvlAllocator {
+  public:
+  TweakedAllocator(
+    CephContext* _cct,
+    int64_t _capacity,
+    int64_t _block_size,
+    std::string_view name)
+    : AvlAllocator( _cct, _capacity, _block_size, name) {
+      }
+  ~TweakedAllocator() override {}
+
+  const char* get_type() const override {
+    return "tweaked";
+  };
+
+  int64_t allocate(
+    uint64_t want_size, uint64_t block_size,
+    uint64_t max_alloc_size, int64_t hint,
+    PExtentVector *extents) override {
+      called_allocate = true;
+      int r = AvlAllocator::allocate(want_size, block_size, max_alloc_size, hint, extents);
+      return r;
+    };
+  bool called_allocate = false;
```

Comment:
Hmm, so all that "allocator creation customization" framework is to be able to track whether allocate() call took place or not only, right? Looks like a bit of overkill to me.
Wouldn't it be more generic and easier approach to introduce some stats counters to Allocator class? E.g. allocate() call counter, release() one ,etc?
Whether these counters should be just integer members in Allocator class or be our regular "public" perf counters (available through ceph tell xxx perf dump) is debatable. IMO having them "public" is nice to have...

Reply:
And once you have relevant counter - it could be used for test case instead of 'called_allocate' flag as well.